include/oxen/quic/types.hpp

```diff
@@ -8,6 +8,25 @@
 
 namespace oxen::quic
 {
+    class Stream;
+    struct Address;
+
+    namespace concepts
```

Comment:
Grouping all of the concepts in this file just because they are concepts feels wrong to me, because the concepts defined here are not generic but rather are specific to types defined elsewhere.  So in other words, I can't actually touch the `stream_derived_type` context defined here without having first included stream.hpp to make `Stream` available, but in that case, why not define it inside stream.hpp?  (This is similar to the STL: `<concepts>` defines highly generic language concepts, with feature-specific ones, such as range-specific concepts, defined alongside the in feature headers, such as `<range>`).